lib/utilities/monitoringHandler.js

```diff
@@ -0,0 +1,65 @@
+const errors = require('arsenal');
+const client = require('prom-client');
+
+function writeResponse(res, error, log, results, cb) {
+    let statusCode = 200;
+    if (error) {
+        if (Number.isInteger(error.code)) {
+            statusCode = error.code;
+        } else {
+            statusCode = 500;
+        }
+    }
+    res.writeHead(statusCode, { 'Content-Type': 'application/json' });
+    res.write(JSON.stringify(results));
+    res.end(() => {
+        cb(error, results);
+    });
+}
+
+
+function routeHandler(req, res, log, cb) {
```

Comment:
In some control flows, `cb` is not called, which is slightly odd.

Reply:
As advised I am following the healthcheckHandler